azurerm/resource_arm_cosmosdb_sql_container.go

```diff
@@ -0,0 +1,215 @@
+	if err != nil {
+		return fmt.Errorf("Error issuing create/update request for Cosmos SQL Container %s (Account: %s, Database:%s): %+v", name, account, database, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+
+		return fmt.Errorf("Error waiting on create/update future for Cosmos SQL Container %s (Account: %s, Database:%s): %+v", name, account, database, err)
+	}
+
+	resp, err := client.GetSQLContainer(ctx, resourceGroup, account, database, name)
+	if err != nil {
+		return fmt.Errorf("Error making get request for Cosmos SQL Container %s (Account: %s, Database:%s): %+v", name, account, database, err)
+	}
+
+	id, err := azure.CosmosGetIDFromResponse(resp.Response)
+	if err != nil {
+		return fmt.Errorf("Error retrieving the ID for Cosmos SQL Container '%s' (Account: %s, Database:%s) ID: %v", name, account, database, err)
+	}
+	d.SetId(id)
+
+	return resourceArmCosmosDbSQLContainerRead(d, meta)
+}
+
+func resourceArmCosmosDbSQLContainerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cosmosAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+	name := d.Get("name").(string)
+
+	id, err := azure.ParseCosmosDatabaseID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.GetSQLContainer(ctx, id.ResourceGroup, id.Account, id.Database, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading Cosmos SQL Container %s (Account %s) - removing from state", id.Database, name)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Cosmos SQL Container %s (Account %s): %+v", id.Database, name, err)
+	}
+
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("account_name", id.Account)
+	d.Set("name", name)
+	d.Set("database_name", id.Database)
+	if props := resp.SQLContainerProperties; props != nil {
+		d.Set("name", props.ID)
+	}
+
+	return nil
+}
+
+func resourceArmCosmosDbSQLContainerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cosmosAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+	name := d.Get("name").(string)
+
+	id, err := azure.ParseCosmosDatabaseID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.DeleteSQLContainer(ctx, id.ResourceGroup, id.Account, id.Database, name)
+	if err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error deleting Cosmos SQL Container %s (Account %s): %+v", id.Database, name, err)
+		}
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting on delete future for Cosmos SQL Container %s (Account %s): %+v", id.Database, id.Account, err)
+	}
+
+	return nil
+}
+
+func expandSQLContainerUniqueKeyPolicy(input interface{}) *[]documentdb.UniqueKey {
+	outputs := make([]documentdb.UniqueKey, 0)
+	var paths []string
+
+	for _, i := range input.(*schema.Set).List() {
+		b := i.(map[string]interface{})
```

Comment:
Could you explain this code? couldn't it instead be:
```
paths = append(paths, b["path"].(string))
```

Reply:
i'll try and see what happens, thanks. Before i was just having issues with the value the Azure SDK is returning.

Reply:
I'm not sure why you are making it an string array pointer, dereferencing it, and then joining it?

`&[]string{b["path"].(string)}` makes a string array of b["path"].(string)
`*` then deferences it
`strings.Join()` joins the array,consisting of just `b["path"].(string)`
leaving a string consisting of `b["path"].(string)` if i'm not mistaken?

Reply:
I've completely redone this now so may have to do a fresh review.